src/librustc_interface/passes.rs

```diff
@@ -350,13 +350,12 @@ fn configure_and_expand_inner<'a>(
         )
     });
 
-    // If we're actually rustdoc then there's no need to actually compile
-    // anything, so switch everything to just looping
-    let mut should_loop = sess.opts.actually_rustdoc;
+    let mut should_loop = false;
```

Comment:
```suggestion
    let should_loop = match sess.opts.pretty { Some(PpMode::PpmSource(PpSourceMode::PpmEveryBodyLoops)) => true, _ => false };
```

(or use the `matches!` macro, if we use that elsewhere in this crate; I cannot recall whether the team has adopted that macro or not.)